### PR TITLE
TagCoverArtImage(): properly set _repr() and _str() for this subclass

### PR DESCRIPTION
**Pull request by zas**
Also add matching tests

# Summary

* This is a…
  * [ ] Bug fix
  * [ ] Feature addition
  * [x] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-XXX

# Solution

# Action

### CODE REVIEW

File: picard/coverart/image.py [hunk 1]
```diff
@@ -469,19 +469,24 @@ def source(self):
...
+            yield 'types=%r' % self.types
         if self.is_front is not None:
-            p.append("is_front=%r" % self.is_front)
-        p.append('support_types=%r' % self.support_types)
+            yield 'is_front=%r' % self.is_front
+        yield 'support_types=%r' % self.support_types
         if self.comment:
-            p.append("comment=%r" % self.comment)
-        return "%s(%s)" % (self.__class__.__name__, ", ".join(p))
+            yield 'comment=%r' % self.comment
```
Comment (phw) [inline picard/coverart/image.py:482]: The parent implementation has a different order for is _front and _support_types and also outputs support_multi_types. I wonder if we should instead call the parent implementation:

```python
def _repr(self):
    yield '%r' % self.sourcefile
    if self.tag is not None:
        yield 'tag=%r' % self.tag
    yield from super()._repr()
```

The only additional field handled by the parent function is `url`, but if that is empty it is skipped.

File: picard/coverart/image.py [hunk 2]
```diff
@@ -469,19 +469,24 @@ def source(self):
...
+
+    def _str(self):
+        yield 'Image from %r' % self.sourcefile
+        if self.types:
+            yield "of type %s" % ','.join(self.types)
+        if self.comment:
+            yield "and comment '%s'" % self.comment
```
Comment (phw) [inline picard/coverart/image.py:489]: Here as well:

```python
def _str(self):
    yield 'Image from %r' % self.sourcefile
    yield from super()._str()
```

**Review by phw — CHANGES_REQUESTED** [review]

File: picard/coverart/image.py (hunk 1 above)
Comment (zas) [reply picard/coverart/image.py]: Done (and I did more or less the same for _str()) in [c8a41de](https://github.com/metabrainz/picard/pull/2319/commits/c8a41de4103e631164241fd81f00ee46f14e4022)

File: picard/coverart/image.py [hunk 3]
```diff
@@ -228,7 +228,6 @@ def __repr__(self):
         return "%s(%s)" % (self.__class__.__name__, ", ".join(self._repr()))
 
     def _str(self):
-        yield 'Image'
```
Comment (phw) [inline picard/coverart/image.py:231]: Not sure what the intention is, but having as string representation just "from /some/file/name" seems odd as well. What about outputting `self.__class__.__name__` here as well?

Comment (zas) [reply picard/coverart/image.py:231]: Done in [6e0054d](https://github.com/metabrainz/picard/pull/2319/commits/6e0054d2423a8d06d2d32d35beca82cf24bd5e04)

**Review by phw — APPROVED** [review]

**Pull request merged**